src/OpenGL/Extensions/Silk.NET.OpenGL.Extensions.ImGui/ImGuiController.cs

```diff
@@ -72,6 +72,27 @@ public ImGuiController(GL gl, IView view, IInputContext input, ImGuiFontConfig i
 
             BeginFrame();
         }
+        
+        /// <summary>
+        /// Constructs a new ImGuiController with an onConfigureAction.
+        /// </summary>
+        public ImGuiController(GL gl, IView view, IInputContext input, Action onConfigureIO)
```

Comment:
Can we change this to

```cs
public ImGuiController(GL gl, IView view, IInputContext input, ImGuiFontConfig? cfg = null, Action? onConfigureIO = null)
```

And implement the old constructor in terms of the new one? That way you're not sacrificing the ability to change the font just so you can do custom IO things (as well as eliminate duplicate code)

Thanks.

Reply:
Well, actually, having a custom action allows user to set any font he/she likes, making the work with ImGui very close to native library. From my pov `ImGuiFontConfig` is a bypass of font configuration, and I would personally remove it, but in terms of back compatibility I better not touch it at all.

Reply:
I personally use the provided API like this:
```csharp
// Our loading function
window.Load += () =>
{
    controller = new ImGuiController(
        gl = editorApp.GL = window.CreateOpenGL(), // load OpenGL
        window, // pass in our window
        inputContext = window.CreateInput(), // create an input context
        () =>
        {
            var io = ImGui.GetIO();
            io.ConfigFlags |= ImGuiConfigFlags.DockingEnable;
            editorApp.openSansFont = io.Fonts.AddFontFromFileTTF("assets/Font/OpenSans-Regular.ttf", 18);
        }
    );
};
```

And later use it in rendering:
```csharp

public void OnImGuiRender()
{
    ImGui.PushFont(openSansFont);
	// Other ImGui stuff
}
```

Reply:
I prefer code paths to be a mix-and-match from a single entrypoint where possible, and this is the philosophy used in other parts of the library.

Users are welcome to pass `null` in for the font config and do their configuration in the callback and vice versa, but if they want to use both there's no reason why we should stop them.

Reply:
Ok, understand, will add in a minute

Reply:
Done, please review